### PR TITLE
Feature :: Domain Step in the Pipeline 

### PR DESCRIPTION
**Pull request by Dalanir**
In order to have the possibility to time travel back to the domain choice, we now consider it like a Step unremovable (I'd like feedback on this, we can maybe improve the check). We had to aslo create a form component in order to change the domain because is no long a dropdown.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ node_modules
 /dist
 coverage
 playground/*.map
+playground/dist/*
```
Comment (adimascio) [inline .gitignore]: Please, remove this line. We do need to track some of the files inside `playground/dist` (e.g. `index.html`). Besides, this change should be made in another PR.

File: src/components/Vqb.vue [hunk 2]
```diff
@@ -70,7 +71,8 @@ export default class Vqb extends Vue {
     this.initialValue = { ...params }; // make a copy
     if (index !== undefined) {
       this.editedStepIndex = index;
-      this.selectStep({ index: index - 1 });
+      const prevIndex = index === 0 ? index : index - 1;
```
Comment (adimascio) [inline src/components/Vqb.vue]: Another way would to do:

```suggestion
      const prevIndex = Math.max(index - 1, 0);
```

Don't know if it's more readable or not.

File: src/components/Vqb.vue [hunk 3]
```diff
@@ -82,13 +84,18 @@ export default class Vqb extends Vue {
     // Reset value from DataViewer
     this.initialValue = undefined;
     const newPipeline: Pipeline = [...this.pipeline];
-    const index = this.computedActiveStepIndex + 1;
+    // FIXME: not sur about that specific implem to handle `domain` step
```
Comment (adimascio) [inline src/components/Vqb.vue]: Small typo in the comment

File: tests/unit/domain-step-form.spec.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    emptyStore = setupStore({});
+  });
+
+  it('should instantiate', () => {
+    const wrapper = shallowMount(DomainStepForm, { store: emptyStore, localVue });
+
+    expect(wrapper.exists()).to.be.true;
+  });
+
+  it('should have exactly one widgetinputtext component', () => {
```
Comment (adimascio) [inline tests/unit/domain-step-form.spec.ts]:
```suggestion
  it('should have exactly one widgetautocomplete component', () => {
```

File: .gitignore (hunk 1 above)
Comment (Dalanir) [reply .gitignore]: Right ! My bad
